client/src/components/AdvisorReports/FilterableAdvisorReportsTable.js

```diff
@@ -24,59 +23,52 @@ const GQL_GET_ADVISOR_REPORTS_INSIGHT = gql`
 `
 
 const DEFAULT_WEEKS_AGO = 3
-const OrganizationAdvisorsTableWithLoader = connect(
-  null,
-  mapDispatchToProps
-)(LoaderHOC("isLoading")("data")(OrganizationAdvisorsTable))
-
-class FilterableAdvisorReportsTable extends Component {
-  static propTypes = {
-    showLoading: PropTypes.func.isRequired,
-    hideLoading: PropTypes.func.isRequired
-  }
-
-  constructor() {
-    super()
-    this.state = {
-      filterText: "",
-      export: false,
-      data: [],
-      isLoading: false,
-      selectedData: []
-    }
-    this.handleFilterTextInput = this.handleFilterTextInput.bind(this)
-    this.handleExportButtonClick = this.handleExportButtonClick.bind(this)
-    this.handleRowSelection = this.handleRowSelection.bind(this)
-  }
 
-  componentDidMount() {
-    this.setState({ isLoading: true })
-    this.props.showLoading()
-    API.query(GQL_GET_ADVISOR_REPORTS_INSIGHT).then(data => {
-      this.setState({
-        isLoading: false,
-        data: data.advisorReportInsights
-      })
-      this.props.hideLoading()
-    })
-  }
-
-  handleFilterTextInput(filterText) {
-    this.setState({ filterText: filterText })
+const FilterableAdvisorReportsTable = props => {
+  const [filterText, setFilterText] = useState("")
+  const [selectedData, setSelectedData] = useState([])
+  const { loading, error, data } = API.useApiQuery(
+    GQL_GET_ADVISOR_REPORTS_INSIGHT
+  )
+  const { done, result } = useBoilerplate({
+    loading,
+    error,
+    ...props
+  })
+  if (done) {
+    return result
   }
 
-  handleExportButtonClick() {
-    let selectedData = this.state.selectedData
-    let allData = this.state.data
-    let exportData = selectedData.length > 0 ? selectedData : allData
-    this.downloadCSV({ data: exportData })
+  const advisorReportInsights = data.advisorReportInsights
+  const handleFilterTextInput = _debounce(setFilterText, 300)
+  const columnGroups = getWeekColumns()
+
+  return (
+    <div>
+      <Toolbar
+        onFilterTextInput={handleFilterTextInput}
+        onExportButtonClick={handleExportButtonClick}
+      />
+      <OrganizationAdvisorsTable
+        data={advisorReportInsights}
+        columnGroups={columnGroups}
+        filterText={filterText}
+        onRowSelection={handleRowSelection}
+      />
+    </div>
+  )
+
+  function handleExportButtonClick() {
+    const exportData =
+      selectedData.length > 0 ? selectedData : advisorReportInsights
+    downloadCSV({ data: exportData })
   }
 
-  handleRowSelection(data) {
-    this.setState({ selectedData: data })
+  function handleRowSelection(data) {
+    setSelectedData(data)
   }
 
-  getWeekColumns() {
+  function getWeekColumns() {
```

Comment:
Maybe it is nicer to have getWeekColumns outside the component.

Reply:
Well, functions that are only used locally can just stay where they are (inside the function that uses them), trying to mimise the scope. It also makes it easier for eslint to spot functions that are unused (and thus can be removed).